### PR TITLE
Added healthz endpoint to OpenShift

### LINKED ISSUE #3346: OpenShift does not support health endpoint robustly
While setting up OpenShift with an external Kubernetes I found that OpenShift `/healthz` would 404 as there was no handler for that endpoint on the pod. 

Currently the OpenShift health endpoint is a simple redirect to the Kube apiserver health check. We should implement a more comprehensive check.

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@derekwaynecarr PTAL

Fixes #3346

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
Is there a test that pings this?

**Comment by stevekuznetsov** [comment]
Test. courtesy of @liggitt:
```
$ openshift start master --write-config=os1
$ openshift start master --write-config=os2 --listen=https://0.0.0.0:8444 --kubeconfig=os1/admin.kubeconfig
$ cp os1/master.etcd-client.* os2/
$ cp os1/serviecaccounts.public.key os2/
$ cat os1/ca.crt >> os2/ca.crt
```

Inside of `os2/master-config.yaml`:
* delete all of the `etcdConfig` block
* append or prepend something to the `kubernetesStoratePrefix` and `openShiftStoragePrefix` to stop name collisions 
* add `serviceaccounts.public.key` to the `publicKeyFiles`

```
$ openshift start master --config=os1/master-config.yaml
$ openshift start master --config=os2/master-config.yaml
$ curl -ik <os2ip>/healthz
```

**Comment by derekwaynecarr** [comment]
I meant automated

**Comment by stevekuznetsov** [comment]
Yeah, unclear how to automate that. Thoughts?

**Comment by liggitt** [comment]
`TestServiceAccountAuthorization` sets up a cluster running against another cluster. Could follow that as an example.

**Comment by liggitt** [comment]
This test should be focused on proxied kube issues (healthz, eventually testing the proxied watch issue, etc)

**Comment by stevekuznetsov** [comment]
@liggitt, right will do

**Comment by stevekuznetsov** [comment]
@liggitt Added test. Passes, but takes ~2m to do so.

File: test/integration/external_kube_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	cluster2MasterConfig.EtcdClientInfo = cluster1MasterConfig.EtcdClientInfo
+	// Set a custom etcd prefix to make sure data is getting sent to cluster1
+	cluster2MasterConfig.EtcdStorageConfig.KubernetesStoragePrefix += "2"
+	cluster2MasterConfig.EtcdStorageConfig.OpenShiftStoragePrefix += "2"
+	// Don't manage any names in cluster2
+	cluster2MasterConfig.ServiceAccountConfig.ManagedNames = []string{}
+	// Don't create any service account tokens in cluster2
+	cluster2MasterConfig.ServiceAccountConfig.PrivateKeyFile = ""
+	// Use the same public keys to validate tokens as cluster1
+	cluster2MasterConfig.ServiceAccountConfig.PublicKeyFiles = cluster1MasterConfig.ServiceAccountConfig.PublicKeyFiles
```
Comment (liggitt) [inline test/integration/external_kube_test.go]: don't run controllers in the second cluster (might speed up start time)
```
cluster2MasterConfig.PauseControllers = true
```

File: test/integration/external_kube_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+	// Start cluster 2 (without clearing etcd) and get admin client configs and clients
+	cluster2Options := testutil.TestOptions{DeleteAllEtcdKeys: false}
+	_, err = testutil.StartConfiguredMasterWithOptions(cluster2MasterConfig, cluster2Options)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Ping the healthz endpoint on the second OpenShift cluster
```
Comment (liggitt) [inline test/integration/external_kube_test.go]: I would expect it to get here in about 15 seconds

**Comment by stevekuznetsov** [comment]
[test]

**Comment by stevekuznetsov** [comment]
@derekwaynecarr Test is in.

**Comment by derekwaynecarr** [comment]
lgtm [merge]

**Comment by liggitt** [comment]
delete all flake, re[merge]

**Comment by stevekuznetsov** [comment]
@liggitt Travis went OOM on `hack/test-go`... again

**Pull request merged**